src/searcher/kiwix-search.cpp

```diff
@@ -28,88 +28,75 @@
+  -v --verbose       Give details about the search process
+  -V --version       Print software version
+  -h --help          Print this help
+)";
+
 
 int main(int argc, char** argv)
 {
-  /* Init the variables */
-  // const char *indexPath =
-  // "/home/itamar/.www.kiwix.org/kiwix/43k0i1j4.default/6d2e587b-d586-dc6a-dc6a-e4ef035a1495d15c.index";
-  // const char *indexPath = "/home/itamar/testindex";
-  const char* zimPath = NULL;
-  const char* search = NULL;
-  bool verboseFlag = false;
-  bool suggestionFlag = false;
-  int option_index = 0;
-  int c = 0;
-
-  /* Argument parsing */
-  while (42) {
-    static struct option long_options[]
-        = {{"verbose", no_argument, 0, 'v'},
-           {"suggestion", no_argument, 0, 's'},
-           {"version", no_argument, 0, 'V'},
-           {0, 0, 0, 0}};
-
-    if (c != -1) {
-      c = getopt_long(argc, argv, "Vvsb:", long_options, &option_index);
-
-      switch (c) {
-        case 'v':
-          verboseFlag = true;
-          break;
-        case 'V':
-          version();
-          return 0;
-        case 's':
-          suggestionFlag = true;
-          break;
-      }
-    } else {
-      if (optind < argc) {
-        if (zimPath == NULL) {
-          zimPath = argv[optind++];
-        } else if (search == NULL) {
-          search = argv[optind++];
-        } else {
-          cout << zimPath << endl;
-          usage();
-        }
-      } else {
-        break;
-      }
-    }
+  Options args;
+  try {
+    args = docopt::docopt_parse(USAGE, {argv+1, argv+argc}, false, false);
+  } catch (docopt::DocoptArgumentError const & error ) {
+    std::cerr << error.what() << std::endl;
+    std::cerr << USAGE << std::endl;
+    return -1;
   }
 
-  /* Check if we have enough arguments */
-  if (zimPath == NULL || search == NULL) {
-    usage();
+  if (args.at("--help").asBool()) {
+    std::cout << USAGE << std::endl;
+    return 0;
   }
 
+  if (args.at("--version").asBool()) {
+    version();
+    return 0;
+  }
+
+  auto zimPath = args.at("ZIM").asString();
+  auto pattern = args.at("PATTERN").asString();
+  auto verboseFlag = args.at("--verbose").asBool();
+
   /* Try to prepare the indexing */
   try {
     zim::Archive archive(zimPath);
 
-    if (suggestionFlag) {
+    if (args.at("--suggestion").asBool()) {
       zim::SuggestionSearcher searcher(archive);
       searcher.setVerbose(verboseFlag);
-      for (const auto& r : searcher.suggest(search).getResults(0, 10) ) {
-        cout << r.getTitle() << endl;
+      auto results = searcher.suggest(pattern).getResults(0, 10);
+      for (auto it=results.begin(); it!=results.end(); it++) {
+        auto suggestionItem = *it;
+        cout << "Title : " << suggestionItem.getTitle() << endl;
+        cout << "Path : " << suggestionItem.getPath() << endl;
+        auto entry = it.getEntry();
+        cout << "Entry title : "  << entry.getTitle() << endl;
+        cout << endl;
```

Comment:
This is a change in the output of `kiwix-search --suggestion`. If it's intentionaly, it better be performed in a separate commit, and I think that the full-text (non `--suggestion`) mode should be updated too.

BTW, when can **"Title :"** be different from **"Entry title :"**?

Reply:
This is not intentionnal, this is a old try/error I had on my working tree.

I will remove this.

Reply:
> BTW, when can "Title :" be different from "Entry title :"?

If, at zim creation, user (scrapper) reimplement a `IndexData::getTitle()` which return something different than entry title (https://github.com/openzim/libzim/blob/main/include/zim/writer/item.h#L73)